### PR TITLE
Switch tiles

### PR DESCRIPTION
**Pull request by malikasen**
- added the functionality of drag and drop. now the tiles can switch places with a neighboring empty cell.

### CODE REVIEW

File: src/App/components/EmptySquare.js [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
 import { switchPositions, canMoveTile } from '../utils';
 
 export default function EmptySquare({
   tilePosition,
-  setShuffledTiles,
-  setIsTilesShuffled,
+  tiles,
+  setTiles,
+  positions,
+  setPositions,
```
Comment (laomatt) [inline src/App/components/EmptySquare.js:11]: unnecessary comma?

File: src/App/components/EmptySquare.js [hunk 2]
```diff
@@ -1,42 +1,49 @@
...
+      accept: ItemTypes.NUMBER,
+      drop: (item) => {
+        switchPositions(
+          tiles[item.tilePosition].props.number,
+          item.tilePosition,
+          tilePosition,
+          tiles,
+          setTiles,
+          positions,
+          setPositions,
```
Comment (laomatt) [inline src/App/components/EmptySquare.js:24]: unnecessary comma?

File: src/App/utils.js [hunk 3]
```diff
@@ -48,30 +50,44 @@ const switchPositions = (
   number,
   oldPosition,
   newPosition,
-  setIsTilesShuffled,
-  setShuffledTiles,
+  tiles,
+  setTiles,
+  positions,
+  setPositions,
```
Comment (laomatt) [inline src/App/utils.js:56]: unnecessary comma?

File: src/App/utils.js [hunk 4]
```diff
@@ -48,30 +50,44 @@ const switchPositions = (
...
-      shuffledTilesAfterSwitch.push(
-        <Numbers number={number} tilePosition={i} />,
-      );
+      tilesAfterSwitch.push(<Numbers number={number} tilePosition={i} />);
     } else if (i === oldPosition) {
-      shuffledTilesAfterSwitch.push(
+      //switching tiles when numbers are shuffled
+      positionsInProgress[newPosition] = {
+        number: number,
+        position: newPosition,
```
Comment (laomatt) [inline src/App/utils.js:67]: unnecessary comma?

File: src/App/utils.js [hunk 5]
```diff
@@ -48,30 +50,44 @@ const switchPositions = (
...
+      };
+      positionsInProgress[oldPosition] = {
+        number: 'empty',
+        position: oldPosition,
```
Comment (laomatt) [inline src/App/utils.js:71]: unnecessary comma?

File: src/App/App.js [hunk 6]
```diff
@@ -9,54 +9,67 @@ import { HTML5Backend } from 'react-dnd-html5-backend';
...
+  const [positions, setPositions] = useState([]);
   const orderTiles = () => {
-    let tilesInProgress = [...tiles];
-    if (tiles.length === 0) {
-      for (let i = 0; i < 15; i++) {
-        tilesInProgress.push(<Numbers number={i + 1} tilePosition={i} />);
-      }
+    const initialPositions = Array.from({ length: 15 }, (_, index) => ({
+      number: index + 1,
+      position: index,
```
Comment (laomatt) [inline src/App/App.js:16]: unnecessary comma?

**Review by laomatt — APPROVED** [review]
just suggestions

File: src/App/App.js (hunk 6 above)
Comment (malikasen) [reply src/App/App.js:16]: added by prettier

**Pull request merged**